pkg/xds/envoy/listeners/v3/fault_injection_configurer.go

```diff
@@ -15,42 +15,48 @@ import (
 )
 
 type FaultInjectionConfigurer struct {
-	FaultInjection *mesh_proto.FaultInjection
+	FaultInjections []*mesh_proto.FaultInjection
 }
 
 func (f *FaultInjectionConfigurer) Configure(filterChain *envoy_listener.FilterChain) error {
-	if f.FaultInjection == nil {
-		return nil
-	}
+	var httpFilters []*envoy_hcm.HttpFilter
 
-	config := &envoy_http_fault.HTTPFault{
-		Delay: convertDelay(f.FaultInjection.Conf.GetDelay()),
-		Abort: convertAbort(f.FaultInjection.Conf.GetAbort()),
-		Headers: []*envoy_route.HeaderMatcher{
-			createHeaders(f.FaultInjection.SourceTags()),
-		},
-	}
+	// Iterate over FaultInjections and generate the relevant HTTP filters.
+	// We do assume that the FaultInjections resource is sorted, so the most
+	// specific source matches come first.
```

Comment:
I think I'm missing something with this comment. My understanding of fault injection is that Envoy applies every matching fault injection filter, not just the first match. Am I wrong here?

Reply:
Probably the comment is slightly convoluted, it just means we have a consistent order based on the source selector. You're right, all matching fault injection filters will be applied.